build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/util/GitUtils.java

```diff
@@ -0,0 +1,143 @@
+package org.jfrog.build.extractor.clientConfiguration.util;
+
+import org.apache.commons.lang.StringUtils;
+import org.jfrog.build.api.Vcs;
+import org.jfrog.build.api.util.Log;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class GitUtils {
+    private static String CREDENTIALS_IN_URL_REGEXP = "((http|https):\\/\\/\\w.*?:\\w.*?@)";
+
+    private static File getDotGit(File file) {
+        if (file == null) {
+            return null;
+        }
+        File dotGit = new File(file, ".git");
+        if (dotGit.exists()) {
+            return dotGit;
+        }
+        return getDotGit(file.getParentFile());
+    }
+
+    /**
+     * Extract Vcs details from .git configuration.
+     */
+    public static Vcs extractVcs(File workingDir, Log log) throws IOException {
+        File dotGit = getDotGit(workingDir);
```

Comment:
I think that a NullPointerException can be thrown here, if there's no .git directory (the *getDotGit* method can return null). I suggest throwing a FileNotFoundException with a clear message here.